sanic/headers.py

```diff
@@ -35,141 +35,78 @@
-        return str.__new__(cls, value)
-
+class MediaType:
+    """A media type, as used in the Accept header."""
     def __init__(
         self,
-        value: str,
-        type_: MediaType,
-        subtype: MediaType,
-        *,
-        q: str = "1.0",
-        **kwargs: str,
+        type_: str,
+        subtype: str,
+        **params: str,
     ):
-        qvalue = float(q)
-        if qvalue > 1 or qvalue < 0:
-            raise InvalidHeader(
-                f"Accept header qvalue must be between 0 and 1, not: {qvalue}"
-            )
-        self.value = value
         self.type_ = type_
         self.subtype = subtype
-        self.qvalue = qvalue
-        self.params = kwargs
-
-    def _compare(self, other, method):
-        try:
-            return method(self.qvalue, other.qvalue)
-        except (AttributeError, TypeError):
-            return NotImplemented
-
-    @parse_arg_as_accept
-    def __lt__(self, other: Union[str, Accept]):
-        return self._compare(other, lambda s, o: s < o)
+        self.q = float(params.get("q", "1.0"))
+        self.params = params
+        self.str = f"{type_}/{subtype}"
 
-    @parse_arg_as_accept
-    def __le__(self, other: Union[str, Accept]):
-        return self._compare(other, lambda s, o: s <= o)
+    def __repr__(self):
+        return self.str + "".join(f";{k}={v}" for k, v in self.params.items())
 
-    @parse_arg_as_accept
-    def __eq__(self, other: Union[str, Accept]):  # type: ignore
-        return self._compare(other, lambda s, o: s == o)
+    def __eq__(self, media_type: str):
+        """Check if the type and subtype match exactly."""
+        return self.str == media_type
 
-    @parse_arg_as_accept
-    def __ge__(self, other: Union[str, Accept]):
-        return self._compare(other, lambda s, o: s >= o)
-
-    @parse_arg_as_accept
-    def __gt__(self, other: Union[str, Accept]):
-        return self._compare(other, lambda s, o: s > o)
-
-    @parse_arg_as_accept
-    def __ne__(self, other: Union[str, Accept]):  # type: ignore
-        return self._compare(other, lambda s, o: s != o)
-
-    @parse_arg_as_accept
     def match(
         self,
-        other,
-        *,
-        allow_type_wildcard: bool = True,
-        allow_subtype_wildcard: bool = True,
-    ) -> bool:
-        type_match = (
-            self.type_ == other.type_
-            if allow_type_wildcard
-            else (
-                self.type_.match(other.type_)
-                and not self.type_.is_wildcard
-                and not other.type_.is_wildcard
-            )
-        )
-        subtype_match = (
-            self.subtype == other.subtype
-            if allow_subtype_wildcard
-            else (
-                self.subtype.match(other.subtype)
-                and not self.subtype.is_wildcard
-                and not other.subtype.is_wildcard
-            )
-        )
-
-        return type_match and subtype_match
+        media_type: str,
```

Comment:
This should accept `str` and `MediaType`

Reply:
This intentionally does not accept MediaTypes, but only mime strings (that have no parameters such as q). But I implemented other types for equality operator because it expects that.

Reply:
This is breaking then.

Reply:
Very obviously. This whole thing depended on `request.accept` never being documented, and getting to .3 release where things can break. This substantially changes the semantics of that handler, not only the innermost type.

Reply:
The most breaking change is probably that of the `in` operator, but `match` gives somewhat different results now. Both were very "broken", and hopefully these changes automatically fix more than they break anything for those users who already discovered the accessor and made use of it.

A particular "oops" moment was with my own app, that had been checking `"text/html" in request.headers.get("accept", "")` (the check itself is a bit b0rked but that's what I had working all these years), and responding in HTML in that case, otherwise plain text (mainly fetch calls from the front, but also some curl users).

I changed that to use `request.accept` instead, hearing of it being added, and suddenly *every client* started matching: fetch and curl accidentally got HTML.

Reply:
I know that we are breaking, but what I thought the extent of the agreed upon change was to allow for a differentiation between exact matching and wildcard matching.

Meaning, that `in` would behave differently than `match`. But it seems that also `match` is changing.